app/models/krikri/activity.rb

```diff
@@ -10,16 +10,41 @@ module Krikri
   class Activity < ActiveRecord::Base
     # @!attribute agent
     #    @return [String] a string representing the Krikri::SoftwareAgent
-    #                     responsible for the activity.
+    #      responsible for the activity.
+    # @!attribute start_time
+    #    @return [DateTime] a datestamp marking the activity's start
     # @!attribute end_time
     #    @return [DateTime] a datestamp marking the activity's competion
     # @!attribute opts
     #    @return [JSON] the options to pass to the #agent class when running
-    #                   the activity
-    # @!attribute start_time
-    #    @return [DateTime] a datestamp marking the activity's start
+    #      the activity
+    
     validate :agent_must_be_a_software_agent
 
+    ##
+    # @example building a valid URI from the base
+    #   Krikri::Activity.base_uri / 1
+    #   
+    # @return [RDF::URI] the configured base URI for this class
+    def self.base_uri
+      RDF::URI.intern(Krikri::Settings['marmotta']['ldp']) /
+        Krikri::Settings['prov']['activity']
+    end
+
+    ##
+    # @param uri [#to_s] a uri for this activity
+    #
+    # @return [Krikri::Activity] the activity with the given uri
+    #
+    # @raise [RuntimeError] if the URI form does not match the activity
+    # @raise [ActiveRecord::RecordNotFound] if no activity is found
+    def self.from_uri(uri)
+      raise "Cannot find #{self} from URI: #{uri}" unless 
+        uri.start_with? base_uri
```

Comment:
Since the condition is that the given URI doesn't start with the base URI, could this message be changed to indicate that the error is with the given URI not matching what we'd like? It hasn't tried to `find` the Activity yet at this point.